src/fraction.ts

```diff
@@ -32,39 +41,44 @@ export class Fraction {
   /**
    * LCM: Lowest common multiple
    */
-  static LCM(a, b) {
+  static LCM(a: number, b: number): number {
     return (a * b) / Fraction.GCD(a, b);
   }
 
   /**
    * LCMM: Lowest common multiple for more than two numbers
    */
-  static LCMM(args) {
+  static LCMM(args: any): number {
     if (args.length === 0) {
       return 0;
-    } else if (args.length === 1) {
+    }
+    if (args.length === 1) {
       return args[0];
-    } else if (args.length === 2) {
+    }
+    if (args.length === 2) {
       return Fraction.LCM(args[0], args[1]);
-    } else {
-      const arg0 = args[0];
-      args.shift();
-      return Fraction.LCM(arg0, Fraction.LCMM(args));
     }
+    const arg0 = args[0];
+    args.shift();
+    return Fraction.LCM(arg0, Fraction.LCMM(args));
   }
 
-  constructor(numerator, denominator) {
-    this.set(numerator, denominator);
+  constructor(numerator?: number, denominator?: number) {
+    this.numerator = numerator === undefined ? 1 : numerator;
```

Comment:
Why not call `this.set` instead of duplicating code?

Reply:
To avoid eslint complaining about not initialising member variables in the constructor. I also do not like the duplication.

Reply:
Can't we instead use:

```
numerator: number = 1;
denominator: number = 1;

constructor(numerator?: number, denominator?: number) {
     this.set(numerator, denominator);
}

set(numerator: number = 1, denominator: number = 1) {
    this.numerator = numerator;
    this.denominator = denominator;
}
```

and if eslint is complaining, let's disable the check?  It seems like we're using a code paradigm that no one likes for the sake of an automated program.  My colleagues use paradigms like this suggested one everywhere and we've never ran into a problem.

Reply:
@mscuthbert very good idea, much clearer